mslib/msui/wms_control.py

```diff
@@ -613,20 +614,27 @@ def on_failure(e):
                         self.cpdlg.close()
                         return
                 else:
-                    self.cpdlg.close()
                     logging.error("cannot load capabilities document.. "
                                   "no layers can be used in this view.")
                     QtWidgets.QMessageBox.critical(
                         self.multilayers, self.tr("Web Map Service"),
                         self.tr(f"ERROR: We cannot load the capability document!\n\n{type(ex)}\n{ex}"))
+                    self.cpdlg.close()
+            except Exception as ex:
+                logging.error("cannot load capabilities document.. "
```

Comment:
maybe refactor the error/box in a local function or reorganise the try/except stuff to get rid of the copy-paste?